service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/gcp/bqdataset/ControlledBigQueryDatasetHandler.java

```diff
@@ -52,4 +54,8 @@ public WsmResource makeResourceFromDb(DbResource dbResource) {
             .build();
     return resource;
   }
+
+  public String generateCloudName(@Nullable UUID workspaceUuid, String bqDatasetName) {
+    return bqDatasetName.replace("-", "_");
```

Comment:
also do a replaceAll("[^A-Z0-9_]", "") as the name does not allow special character except for underscore Letters (uppercase or lowercase), numbers, and underscores. and trim off leading "_".

Reply:
+1, if this method is expected to return a valid cloud resource name then we should clean up/remove special characters/cut off after max length (which should be 1024 characters)